Fixing PSSA errors (#92,#93,#70)

IngoGege opened the pull request:
Fixed PSSA issues #92 
No output for MSFT_xExchWaitForMailboxDatabase and excluding:
PSDSCDscExamplesPresent
PSDSCDscTestsPresent

These changes fix the few PSSA issues that came up in the
MSFT_xExchWebServicesVirtualDirectory resource. (#93)

There is no longer any output when running:
Invoke-ScriptAnalyzer
.\DSCResources\MSFT_xExchWebServicesVirtualDirectory\ -ExcludeRule
@('PSDSCDscExamplesPresent', 'PSDSCDscTestsPresent')

These changes fix the few PSSA issues that came up in the
MSFT_xExchExchangeCertificate resource. (#70)

There is no longer any output when running:
Invoke-ScriptAnalyzer .\DSCResources\MSFT_xExchExchangeCertificate\ -ExcludeRule @('PSDSCDscExamplesPresent', 'PSDSCDscTestsPresent')

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/xexchange/100)
<!-- Reviewable:end -->

kwirkykat commented:
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[DSCResources/MSFT_xExchExchangeCertificate/MSFT_xExchExchangeCertificate.psm1, line 313 \[r1\]](https://reviewable.io:443/reviews/powershell/xexchange/100#-KIE8zdmAoC3BjauiJiN:-KIE8zdnD2nUwyOvPHZe:150124380) ([raw file](https://github.com/powershell/xexchange/blob/25172a77ae36ebb8cf1b5bfe2a41738a17f92670/DSCResources/MSFT_xExchExchangeCertificate/MSFT_xExchExchangeCertificate.psm1#L313)):*
> ```PowerShell
> function CompareCertServices
> {
>     [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSProvideCommentHelp", "")]
> ```

This rule can be fixed rather than suppressed. The comments are already here. They just aren't formatted correctly. If you change the comments already provided to use the help comment format <# .SYNOPSIS ... #> format rather than the # # # on each line format, that will satisfy the rule.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xexchange/100)*
<!-- Sent from Reviewable.io -->

IngoGege commented:
Hi Katie,



I fixed that and removed „[Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSProvideCommentHelp", "")]“. Just wondering no complains for the function GetExchangeCertificate as this one has also no help comment.



Ciao,

Ingo



From: Katie Keim [mailto:notifications@github.com]
Sent: Freitag, 20. Mai 2016 19:26
To: PowerShell/xExchange <xExchange@noreply.github.com>
Cc: Gegenwarth, Ingo <ingo.gegenwarth@sap.com>; Author <author@noreply.github.com>
Subject: Re: [PowerShell/xExchange] Fixing PSSA errors (#93,#70) (#100)



Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

  _____

DSCResources/MSFT_xExchExchangeCertificate/MSFT_xExchExchangeCertificate.psm1, line 313 [r1]<https://reviewable.io:443/reviews/powershell/xexchange/100#-KIE8zdmAoC3BjauiJiN:-KIE8zdnD2nUwyOvPHZe:150124380> (raw file<https://github.com/powershell/xexchange/blob/25172a77ae36ebb8cf1b5bfe2a41738a17f92670/DSCResources/MSFT_xExchExchangeCertificate/MSFT_xExchExchangeCertificate.psm1#L313>):

   function CompareCertServices
   {
       [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSProvideCommentHelp", "")]

   This rule can be fixed rather than suppressed. The comments are already here. They just aren't formatted correctly. If you change the comments already provided to use the help comment format <# .SYNOPSIS ... #> format rather than the # # # on each line format, that will satisfy the rule.

     _____

   Comments from Reviewable<https://reviewable.io:443/reviews/powershell/xexchange/100>

   —
   You are receiving this because you authored the thread.
   Reply to this email directly or view it on GitHub<https://github.com/PowerShell/xExchange/pull/100#issuecomment-220667636> <https://github.com/notifications/beacon/ANqoTiDv5rhtg3Gw7wEXGYR7-z6rffaeks5qDe6ngaJpZM4Ii7C7.gif>

kwirkykat commented:
Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[DSCResources/MSFT_xExchExchangeCertificate/MSFT_xExchExchangeCertificate.psm1, line 310 \[r2\]](https://reviewable.io:443/reviews/powershell/xexchange/100#-KIUsrU7r-g1ulVr1lHb:-KIUsrU7r-g1ulVr1lHc:1914583514) ([raw file](https://github.com/powershell/xexchange/blob/c0ccfa334581c238f7a2bc8bc9ffbb8b0b090ccd/DSCResources/MSFT_xExchExchangeCertificate/MSFT_xExchExchangeCertificate.psm1#L310)):*
> ```PowerShell
> function CompareCertServices
> {
> <#
> ```

Per the [DSC Resource style guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#support-comment-based-help), can you please move the help comment to above the 'function' keyword?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xexchange/100)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
GetExchangeCertificate does not trigger the PSProvideCommentHelp rule here because this rule only checks exported functions.

CompareCertServices is exported as a function from the module at the end of the file.

kwirkykat commented:
@IngoGege Can you also please add your changes to the README?

IngoGege commented:
Hi Katie,
sorry for the delay. It’s now fixed. I had it before above, but I still got the warning, therefore I moved it within the function. Seems like my session caused this issue…
Ciao,
Ingo

From: Katie Keim [mailto:notifications@github.com]
Sent: Mittwoch, 25. Mai 2016 23:35
To: PowerShell/xExchange <xExchange@noreply.github.com>
Cc: Gegenwarth, Ingo <ingo.gegenwarth@sap.com>; Mention <mention@noreply.github.com>
Subject: Re: [PowerShell/xExchange] Fixing PSSA errors (#93,#70) (#100)


@IngoGege<https://github.com/IngoGege> Can you also please add your changes to the README?

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/PowerShell/xExchange/pull/100#issuecomment-221715064>

kwirkykat commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

<details><summary><i>Previously, IngoGege wrote…</i></summary>
> Hi Katie, sorry for the delay. It’s now fixed. I had it before above, but I still got the warning, therefore I moved it within the function. Seems like my session caused this issue… Ciao, Ingo
> 
> From: Katie Keim [mailto:notifications@github.com] Sent: Mittwoch, 25. Mai 2016 23:35 To: PowerShell/xExchange <xExchange@noreply.github.com> Cc: Gegenwarth, Ingo <ingo.gegenwarth@sap.com>; Mention <mention@noreply.github.com> Subject: Re: [PowerShell/xExchange] Fixing PSSA errors (#93,#70) (#100)
> 
> 
> @IngoGege<https://github.com/IngoGege> Can you also please add your changes to the README?
> 
> — You are receiving this because you were mentioned. Reply to this email directly or view it on GitHub<https://github.com/PowerShell/xExchange/pull/100#issuecomment-221715064> 
</details>


---

Reviewed 1 of 1 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xexchange/100#-:-KJBxbBL1NDJxsw2BdyZ:1417088721)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
I can add your changes to the README since my PRs (#101, #102) will change the same bullet of the README as well.
